File: dbassistant/layers/EvaluationLayer.py
from dbassistant.interfaces.ILayer import ILayer
from dbassistant.neural.QueryGenerator import QueryGenerator
from dbassistant.tokens.TokenSeries import TokenSeries


class EvaluationLayer(ILayer):
    def __init__(self, query_generator: QueryGenerator):
        self.query_generator = query_generator

    def process(self, cargo):
        input = cargo.take()
        assert isinstance(input, TokenSeries)
        output = self.query_generator.evaluate(input)
        cargo.put(output)
        return cargo

File: dbassistant/layers/EvaluationLayer.ts
import ILayer from '../interfaces/ILayer';
import QueryGenerator from '../neural/QueryGenerator';
import TokenSeries from '../tokens/TokenSeries';
import LayerCargo from './LayerCargo';

export default class EvaluationLayer implements ILayer {
    private query_generator: QueryGenerator;

    constructor(query_generator: QueryGenerator) {
        this.query_generator = query_generator;
    }

    process(cargo: LayerCargo): LayerCargo {
        const input = cargo.take();
        if (!(input instanceof TokenSeries)) {
            throw new Error('Input must be an instance of TokenSeries');
        }

        const output = this.query_generator.evaluate(input);
        cargo.put(output);
        return cargo;
    }
}
